File: iosApp/iosApp/ViewModels/ItemDetailViewModel.swift
import Foundation
import SharedLogic
import Combine

@MainActor
class ItemDetailViewModel: ObservableObject {
    @Published var item: PantryItem
    @Published var isLoading = false
    @Published var toast: Toast? = nil
    
    private let useCaseHelper = UseCaseHelper()
    private let itemId: String

    init(item: PantryItem) {
        self.item = item
        self.itemId = item.id
    }
    
    func refresh() async {
        isLoading = true
        do {
            self.item = try await useCaseHelper.getItemDetail(itemId: itemId)
        } catch {
            self.toast = Toast(message: "Failed to refresh item", type: .error)
        }
        isLoading = false
    }

    func markAsUsed() {
        // Logic to decrease quantity
    }
    
    func increaseQuantity() {
        // Logic to increase quantity
    }
}

File: sharedUI/src/commonMain/kotlin/com/iqbalfauzi/kitchenstockmanager/presentation/detail/ItemDetailViewModel.kt
package com.iqbalfauzi.kitchenstockmanager.presentation.detail

import androidx.lifecycle.viewModelScope
import com.iqbalfauzi.kitchenstockmanager.domain.usecase.GetItemDetailUseCase
import com.iqbalfauzi.kitchenstockmanager.presentation.BaseViewModel
import kotlinx.coroutines.launch

class ItemDetailViewModel(
    private val itemId: String,
    private val getItemDetailUseCase: GetItemDetailUseCase
) : BaseViewModel<ItemDetailState, ItemDetailIntent, ItemDetailEffect>(ItemDetailState()) {

    init {
        handleIntent(ItemDetailIntent.LoadItem(itemId))
    }

    override fun handleIntent(intent: ItemDetailIntent) {
        when (intent) {
            is ItemDetailIntent.LoadItem -> loadItem(intent.itemId)
            is ItemDetailIntent.Refresh -> loadItem(itemId)
            is ItemDetailIntent.MarkAsUsed -> { /* Mark as used logic */ }
            is ItemDetailIntent.IncreaseQuantity -> { /* Increase quantity logic */ }
        }
    }

    private fun loadItem(id: String) {
        viewModelScope.launch {
            setState { copy(isLoading = true) }
            getItemDetailUseCase(id).fold(
                onSuccess = { item ->
                    setState { copy(isLoading = false, item = item) }
                },
                onFailure = { error ->
                    setState { copy(isLoading = false, error = error.message) }
                    setEffect(ItemDetailEffect.ShowError(error.message ?: "Unknown error"))
                }
            )
        }
    }
}
